clients/pkg/promtail/client/client.go

```diff
@@ -174,12 +178,13 @@ func newClient(metrics *Metrics, cfg Config, streamLagLabels []string, logger lo
 	ctx, cancel := context.WithCancel(context.Background())
 
 	c := &client{
-		logger:          log.With(logger, "component", "client", "host", cfg.URL.Host),
-		cfg:             cfg,
-		entries:         make(chan api.Entry),
-		metrics:         metrics,
-		streamLagLabels: streamLagLabels,
-		name:            asSha256(cfg),
+		logger:              log.With(logger, "component", "client", "host", cfg.URL.Host),
+		cfg:                 cfg,
+		entries:             make(chan api.Entry),
+		metrics:             metrics,
+		streamLagMetricsMap: make(map[string]prometheus.Labels),
```

Comment:
I think we need this new map to be behind a mutex to avoid panics with getting/setting a hashmap concurrently.